src/backend/core/tests/rooms/test_api_rooms_lobby.py

```diff
@@ -0,0 +1,401 @@
+    assert response.json()["status"] == "accepted"
+    assert response.json()["livekit"] == {"token": "test-token"}
+
+
+def test_request_entry_invalid_data():
+    """Should return 400 for invalid request data."""
+    room = RoomFactory()
+    client = APIClient()
+
+    response = client.post(
+        f"/api/v1.0/rooms/{room.id}/request-entry/",
+        {},  # Missing required username field
+    )
+
+    assert response.status_code == 400
+
+
+def test_request_entry_room_not_found():
+    """Should return 404 for non-existent room."""
+    client = APIClient()
+
+    response = client.post(
+        "/api/v1.0/rooms/00000000-0000-0000-0000-000000000000/request-entry/",
+        {"username": "anonymous"},
+    )
+
+    assert response.status_code == 404
+
+
+# Tests for allow_participant_to_enter endpoint
+
+
+def test_allow_participant_to_enter_anonymous():
+    """Anonymous users should not be allowed to manage entry requests."""
+    room = RoomFactory()
+    client = APIClient()
+
+    response = client.post(
+        f"/api/v1.0/rooms/{room.id}/enter/",
+        {"participant_id": "test-id", "allow_entry": True},
+    )
+
+    assert response.status_code == 401
+
+
+def test_allow_participant_to_enter_non_owner():
+    """Non-privileged users should not be allowed to manage entry requests."""
+    room = RoomFactory()
+    user = UserFactory()
+    client = APIClient()
+    client.force_login(user)
+
+    response = client.post(
+        f"/api/v1.0/rooms/{room.id}/enter/",
+        {"participant_id": "test-id", "allow_entry": True},
+    )
+
+    assert response.status_code == 403
+
+
+def test_allow_participant_to_enter_public_room():
+    """Should return 404 for public rooms that don't use the lobby system."""
+    room = RoomFactory(access_level=RoomAccessLevel.PUBLIC)
+    user = UserFactory()
+    # Make user the room owner
+    room.accesses.create(user=user, role="owner")
+
+    client = APIClient()
+    client.force_login(user)
+
+    response = client.post(
+        f"/api/v1.0/rooms/{room.id}/enter/",
+        {"participant_id": "test-id", "allow_entry": True},
+    )
+
+    assert response.status_code == 404
+    assert response.json() == {"message": "Room has no lobby system."}
+
+
+def test_allow_participant_to_enter_success(
+    mock_lobby_service_class, mock_lobby_service
+):
+    """Should successfully update participant status when everything is correct."""
+    room = RoomFactory(access_level=RoomAccessLevel.RESTRICTED)
+    user = UserFactory()
+    # Make user the room owner
+    room.accesses.create(user=user, role="owner")
+
+    client = APIClient()
+    client.force_login(user)
+
+    response = client.post(
+        f"/api/v1.0/rooms/{room.id}/enter/",
+        {"participant_id": "test-id", "allow_entry": True},
+    )
+
+    mock_lobby_service.handle_participant_entry.assert_called_once_with(
+        room_id=room.id,
+        participant_id="test-id",
+        allow_entry=True,
+    )
```

Comment:
IMO you can do the same thing without this mock. Before sending the request you add in the cache the participant in waiting status.
After the request made, the status in the cache should have changed.
You test everything and your test description will be more accurate. You are "just" asserting that the method `handle_participant_entry` has been called. But not that the status has changed.